src/tribol/mesh/CouplingScheme.hpp

```diff
@@ -645,6 +650,51 @@ class CouplingScheme {
    */
   RealT getEffectiveBinningProximityScale() const { return m_effective_binning_proximity_scale; }
 
+#ifdef TRIBOL_USE_ENZYME
+  /**
+   * @brief Enables Enzyme AD for exact Jacobian calculations
+   *
+   * @param useEnzyme Turns on Enzyme support if true
+   */
+  void enableEnzyme( bool useEnzyme ) { m_useEnzyme = useEnzyme; }
+
+  /**
+   * @brief Is Enzyme AD enabled for exact Jacobian calculations?
+   *
+   * @return true Yes
+   * @return false No
+   */
+  bool isEnzymeEnabled() const { return m_useEnzyme; }
+
+  /**
+   * @brief Set nodal normal computation method
+   *
+   * @param nodalNormal NodalNormal object
+   */
+  void initNodalNormal( std::unique_ptr<NodalNormal>&& nodalNormal ) { m_nodalNormal = std::move( nodalNormal ); }
+
+  /**
+   * @brief Get pointer to the NodalNormal object
+   *
+   * @return NodalNormal*
+   */
+  NodalNormal* getNodalNormal() { return m_nodalNormal.get(); }
+
+  /**
+   * @brief Create MethodData to save Jacobian contributions related to a
+   * nodally defined normal
+   */
+  void createNormalJacobian() { m_dnJacobian = std::make_unique<MethodData>(); }
```

Comment:
move to method data that lives on the coupling scheme along with any other method specific normal information

Reply:
`MethodData` is basically a class that stores a Jacobian, so it's most natural to create new `MethodData`s to store other global Jacobian contributions.  Refactoring Jacobian storage is probably outside the scope of this PR, what do you think?